pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -875,6 +937,44 @@ func (gce *GCECloud) firewallNeedsUpdate(name, serviceName, region, ipAddress st
 	return true, false, nil
 }
 
+func (gce *GCECloud) ensureHttpHealthCheckFirewall(region string, hosts []*gceInstance, hcName string, hcPort int32, isNodesHealthCheck bool) error {
+	// Prepare the firewall params for creating / checking.
+	desc := "Firewall for health checks from load-balancer."
+	sourceRanges := lbSrcRngsFlag.ipn
+	ports := []v1.ServicePort{{Protocol: "tcp", Port: hcPort}}
+
+	clusterId, err := gce.ClusterId.GetId()
+	if err != nil {
+		return fmt.Errorf("error getting cluster ID: %v", err)
+	}
+	fwNameSuffix := MakeHealthCheckFirewallNameSuffix(clusterId, hcName, isNodesHealthCheck)
+	fwName := makeFirewallName(fwNameSuffix)
```

Comment:
Regarding the use of `k8s-fw`... this is actually referring to the forwarding rule for an ingress obj. 

From our discussions with Tim, I think the desire is to get rid of these segments. Ideally, clusterId is always the second field of the string. `k8s-clusterid-....`
Would it be possible to adopt the naming scheme that I'm using?  I haven't looked at your other names just yet.
https://github.com/nicksardo/kubernetes/blob/gke-ilb/pkg/cloudprovider/providers/gce/gce_loadbalancer_names.go
Any comments on it?